tests/e2e_tests/utils.py

```diff
@@ -239,3 +242,195 @@ def extract_archive(archive_file: Path, extract_to: Path, cleanup_archive: bool
         print(f"Cleaned up archive: {archive_file}")
 
     return extract_to
+
+
+def extract_compose_container_info(container: docker.models.containers.Container,
+                                    project_name: str) -> QdrantContainer:
+    """Extract container info from a docker-compose container.
+
+    Args:
+        container: Docker container object from docker-compose
+        project_name: Docker-compose project name
+
+    Returns:
+        QdrantContainer: Container info object with compose_project set
+
+    Raises:
+        RuntimeError: If HTTP port mapping cannot be found
+    """
+    container.reload()
+
+    port_bindings = container.attrs['NetworkSettings']['Ports']
+
+    # Find HTTP port (6333)
+    http_port = None
+    if '6333/tcp' in port_bindings and port_bindings['6333/tcp']:
+        http_port = int(port_bindings['6333/tcp'][0]['HostPort'])
+    else:
+        # Look for any exposed HTTP port
+        for port_key, bindings in port_bindings.items():
+            if bindings and port_key.endswith('/tcp'):
+                port_num = int(port_key.split('/')[0])
+                if 6000 <= port_num <= 7000:  # Reasonable range for Qdrant
+                    http_port = int(bindings[0]['HostPort'])
+                    break
+
+    if not http_port:
+        raise RuntimeError(f"Could not find HTTP port mapping for container {container.name}")
+
+    # Find gRPC port (6334) - optional
+    grpc_port = None
+    if '6334/tcp' in port_bindings and port_bindings['6334/tcp']:
+        grpc_port = int(port_bindings['6334/tcp'][0]['HostPort'])
+
+    return QdrantContainer(
+        container=container,
+        host="127.0.0.1",
+        name=container.name,
+        http_port=http_port,
+        grpc_port=grpc_port,
+        compose_project=project_name
+    )
+
+
+def run_docker_compose(docker_client, qdrant_image, test_data_dir, config):
```

Comment:
Can we return something like `QdrantCluster` here? It can have a dedicated clean up method that can be called. Wdyt?